includes/plugins/woocommerce-subscriptions/class-subscriptions-tiers.php

```diff
@@ -456,5 +606,53 @@ public static function order_button_text( $text ) {
 		}
 		return $text;
 	}
+
+	/**
+	 * Render primary product modal.
+	 */
+	public static function print_primary_product_modal() {
+		$query_param = self::get_upgrade_subscription_query_param();
+		if ( empty( $_GET[ $query_param ] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return;
+		}
+
+		$product = self::get_primary_subscription_tier_product();
+		if ( ! $product ) {
+			return;
+		}
+
+		$user_subscription = self::get_user_subscription( $product );
+		$switch_data       = null;
+
+		if ( $user_subscription ) {
+			$product_id = $product->get_id();
+			$item       = null;
+			foreach ( $user_subscription->get_items() as $line_item ) {
+				if (
+					$line_item['product_id'] === $product_id
+					|| $line_item['variation_id'] === $product_id
+					|| ( method_exists( $product, 'get_children' ) && in_array( $line_item['product_id'], $product->get_children() ) ) // In case it's a grouped product.
+				) {
+					$item = $line_item;
+					break;
+				}
+			}
+			if ( $item ) {
+				$switch_data = [
+					'item_id'      => $item['id'],
```

Comment:
I was having trouble with getting the checkout modal to open from the "Change subscription" version of the modal until I changed it to this:

```suggestion
					'item_id'      => $item->get_id(),
```